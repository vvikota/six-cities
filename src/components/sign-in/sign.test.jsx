import React from 'react';
import renderer from 'react-test-renderer';
import SignIn from './sign-in.jsx';
import {MemoryRouter} from "react-router-dom";

it(`SignIn is correctly renderer`, ()=> {
  const tree = renderer.create(<MemoryRouter><SignIn
    logIn={jest.fn()}
    onChange={jest.fn()}
    password={`test`}
    email={`test@test.ru`}
  /></MemoryRouter>).toJSON();

  expect(tree).toMatchSnapshot();
});
