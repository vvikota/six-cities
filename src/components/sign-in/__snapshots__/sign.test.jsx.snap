// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn is correctly renderer 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
      </div>
    </div>
  </header>,
  <main
    className="page__main page__main--login"
  >
    <div
      className="page__login-container container"
    >
      <section
        className="login"
      >
        <h1
          className="login__title"
        >
          Sign in
        </h1>
        <form
          action="#"
          className="login__form form"
          method="post"
        >
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              E-mail
            </label>
            <input
              className="login__input form__input"
              name="email"
              onChange={[MockFunction]}
              placeholder="Email"
              required=""
              type="email"
              value="test@test.ru"
            />
          </div>
          <div
            className="login__input-wrapper form__input-wrapper"
          >
            <label
              className="visually-hidden"
            >
              Password
            </label>
            <input
              className="login__input form__input"
              name="password"
              onChange={[MockFunction]}
              placeholder="Password"
              required=""
              type="password"
              value="test"
            />
          </div>
          <button
            className="login__submit form__submit button"
            onClick={[MockFunction]}
            type="submit"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              Sign in 
            </a>
          </button>
        </form>
      </section>
      <section
        className="locations locations--login locations--current"
      >
        <div
          className="locations__item"
        >
          <a
            className="locations__item-link"
            href="#"
          >
            <span>
              Amsterdam
            </span>
          </a>
        </div>
      </section>
    </div>
  </main>,
]
`;
